src/legacy/core_plugins/telemetry/server/get_telemetry_opt_in.ts

```diff
@@ -43,4 +43,16 @@ export async function getTelemetryOptIn(request: any) {
 
     throw error;
   }
+
+  const { attributes } = savedObject;
+  if (attributes.enabled !== false) return attributes.enabled;
+
+  // Additional check if they've already opted out (enabled: false) - if the
+  // Kibana version has changed since they opted out, set them back to the null
+  // state to get prompted again.
+  const config = request.server.config();
+  const kibanaVersion = config.get('pkg.version');
+  if (kibanaVersion !== attributes.lastVersionChecked) return null;
```

Comment:
I might be wrong here, but won't this cause us to ask the question on every patch version too?